cmake/JoinPaths.cmake

```diff
@@ -0,0 +1,22 @@
+# This module provides function for joining paths
+# known from most languages
+#
+# SPDX-License-Identifier: (MIT OR CC0-1.0)
+# Copyright 2020 Jan Tojnar
+# https://github.com/jtojnar/cmake-snips
+#
+# Modelled after Python’s os.path.join
+# https://docs.python.org/3.7/library/os.path.html#os.path.join
+function(join_paths joined_path first_path_segment)
+  set(temp_path "${first_path_segment}")
+  foreach(current_segment IN LISTS ARGN)
+    if(NOT ("${current_segment}" STREQUAL ""))
+      if(IS_ABSOLUTE "${current_segment}")
+        set(temp_path "${current_segment}")
+      else()
+        set(temp_path "${temp_path}/${current_segment}")
+      endif()
+    endif()
+  endforeach()
+  set(${joined_path} "${temp_path}" PARENT_SCOPE)
```

Comment:
Since the path separators here are specific to unix-y environments, could we try to ["fix" the joined path](https://cmake.org/cmake/help/latest/command/file.html#to-native-path) prior to return? Perhaps, something like: 
```
file(TO_NATIVE_PATH "${joined_path}" joined_path)
```

Reply:
I was also wondering how I could add Windows support - thanks!